### PR TITLE
User Guide: Combine CLI pages

### PR DESCRIPTION
**Pull request by andygrove**
# Which issue does this PR close?

Closes https://github.com/apache/arrow-datafusion/issues/1352

 # Rationale for this change

Why have two mostly duplicate pages documenting the CLI when you can have just one?

# What changes are included in this PR?

Combine pages and fix formatting

# Are there any user-facing changes?

No

### CODE REVIEW

**Comment by kmitchener** [comment]
This closes #1352 actually

File: docs/source/user-guide/cli.md [hunk 1]
```diff
@@ -47,31 +33,125 @@ $ echo "1,2" > data.csv
 
 ```bash
 $ datafusion-cli
+DataFusion CLI v11.0.0
```
Comment (alamb) [inline docs/source/user-guide/cli.md:36]: `11.0` has a nice ring to it!

**Review by alamb — APPROVED** [review]
I love it -- looks great thank you @andygrove

File: docs/source/user-guide/cli.md [hunk 2]
```diff
@@ -47,31 +33,125 @@ $ echo "1,2" > data.csv
...
-+---+---+
-| 1 | 2 |
-+---+---+
-1 row in set. Query took 0.017 seconds.
+### Install and run using Homebrew (on MacOS)
+
+DataFusion CLI can also be installed via Homebrew (on MacOS). Install it as any other pre-built software like this:
+
+```bash
+brew install datafusion
```
Comment (alamb) [inline docs/source/user-guide/cli.md:59]: TIL the homebrew formuale is still updated 🎉  10.0.0 👍 

```bash
alamb@MacBook-Pro-8 datafusion-cli % brew install datafusion
Running `brew update --auto-update`...
==> Auto-updated Homebrew!
Updated 3 taps (homebrew/core, homebrew/cask and homebrew/services).
==> New Formulae
cargo-crev            crytic-compile        git-sync              lucky-commit          slither-analyzer
censys                gcc@11                go@1.18               ouch                  solc-select
cql-proxy             gcem                  libvatek              rush-parallel         svt-av1
==> New Casks
aliwangwang           imhex                 moderndeck            playcover-community   yubihsm2-sdk
gittyup               micromamba            nanoem                virtualbuddy

You have 28 outdated formulae installed.
You can upgrade them with brew upgrade
or list them with brew outdated.

==> Downloading https://ghcr.io/v2/homebrew/core/datafusion/manifests/10.0.0
######################################################################## 100.0%
==> Downloading https://ghcr.io/v2/homebrew/core/datafusion/blobs/sha256:030d1c76aabdc29cbb29c8538512ea5dd6e051
==> Downloading from https://pkg-containers.githubusercontent.com/ghcr1/blobs/sha256:030d1c76aabdc29cbb29c85385
######################################################################## 100.0%
==> Pouring datafusion--10.0.0.monterey.bottle.tar.gz
🍺  /usr/local/Cellar/datafusion/10.0.0: 9 files, 28.9MB
==> Running `brew cleanup datafusion`...
```

File: docs/source/user-guide/cli.md [hunk 3]
```diff
@@ -47,31 +33,125 @@ $ echo "1,2" > data.csv
...
-Build the `datafusion-cli`:
+Use the following commands to clone this repository and build a Docker image containing the CLI tool. Note
+that there is `.dockerignore` file in the root of the repository that may need to be deleted in order for
+this to work.
 
 ```bash
-cd arrow-datafusion/datafusion-cli
-cargo build
+git clone https://github.com/apache/arrow-datafusion
+git checkout 8.0.0
```
Comment (alamb) [inline docs/source/user-guide/cli.md:82]: It might make sense to update this version

**Comment by alamb** [comment]
Thanks @kmitchener  for the assist

**Pull request merged**